### PR TITLE
Discovery ops

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
# Description
Discovery Resource - to discovery device from ome.

# ISSUE TYPE
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### RESOURCE OR DATASOURCE NAME
ome_discovery

##### OUTPUT

```
root@lglw9915:~/ome-discover/terraform-provider-ome# /root/sdk/go1.20.6/bin/go test -coverprofile=coverage.out -timeout 90m -run TestDiscovery terraform-provider-ome/ome
ok      terraform-provider-ome/ome      12.064s coverage: 12.0% of statements
root@lglw9915:~/ome-discover/terraform-provider-ome# go tool cover -html=coverage.out -o coverage.html
```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

**Comment by sachin-apa** [comment]
run e2e

File: docs/resources/discovery.md [hunk 1]
```diff
@@ -40,25 +40,19 @@ Resource for managing discovery on OpenManage Enterprise.
 
 ### Optional
 
-- `community_types_string` (Boolean) - Enable the use of SNMP community strings to receive SNMP traps using Application Settings in OpenManage Enterprise. 
+- `community_string` (Boolean) - Enable the use of SNMP community strings to receive SNMP traps using Application Settings in OpenManage Enterprise.
```
Comment (rounak-adhikary) [inline docs/resources/discovery.md:43]: I think the name "community_string" for a boolean value is unintuitive. Please check if the name can be made better.

File: ome/resource_discovery.go [hunk 2]
```diff
@@ -48,6 +53,74 @@ func (r *discoveryResource) Schema(_ context.Context, _ resource.SchemaRequest,
...
+				path.Root("cron"),
+				"Attribute Error",
+				"With Schedule as RunNow, CRON can't be set.",
+			)
+		}
+	}
+
+	if len(data.DiscoveryConfigTargets) > 0 {
+		for idx, dct := range data.DiscoveryConfigTargets {
+			idxError := "Inappropriate value for attribute `discovery_config_targets`: element: " + strconv.Itoa(idx) + "\n"
```
Comment (rounak-adhikary) [inline ome/resource_discovery.go:75]: Quotes are preferred over backticks in errors.

File: ome/resource_discovery.go [hunk 3]
```diff
@@ -48,6 +53,74 @@ func (r *discoveryResource) Schema(_ context.Context, _ resource.SchemaRequest,
...
+					idxError+"Atleast one of device type should be configured. ")
+			}
+			if len(dct.NetworkAddressDetail) == 0 {
+				resp.Diagnostics.AddAttributeError(
+					path.Root("discovery_config_targets").AtListIndex(idx).AtName("network_address_detail"),
+					"Attribute Error",
+					idxError+"Atleast one of network address detail should be configured. ")
+			}
+		}
+	} else {
```
Comment (rounak-adhikary) [inline ome/resource_discovery.go:105]: These length validations can be done using https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.11.0/listvalidator#SizeAtLeast, rt?

File: ome/resource_discovery.go [hunk 4]
```diff
@@ -114,6 +213,29 @@ func (r *discoveryResource) Update(ctx context.Context, req resource.UpdateReque
 		return
 	}
 
+	if !reflect.DeepEqual(state, plan) {
```
Comment (rounak-adhikary) [inline ome/resource_discovery.go:216]: Since u are in the update procedure, will this condition ever be false?

**Comment by ncsuman7** [comment]
run e2e

File: docs/resources/discovery.md (hunk 1 above)
Comment (Krishnan-Priyanshu) [reply docs/resources/discovery.md:43]: does renaming it to "enable_community_strings" will be intuitive?

File: ome/resource_discovery.go (hunk 2 above)
Comment (Krishnan-Priyanshu) [reply ome/resource_discovery.go:75]: replacing ` to \"

File: docs/resources/discovery.md (hunk 1 above)
Comment (rounak-adhikary) [reply docs/resources/discovery.md:43]: Yes, sounds good

File: ome/resource_discovery.go (hunk 4 above)
Comment (Krishnan-Priyanshu) [reply ome/resource_discovery.go]: removed the conditions

File: ome/resource_discovery.go (hunk 3 above)
Comment (Krishnan-Priyanshu) [reply ome/resource_discovery.go]: added the list validators.

File: ome/resource_discovery.go (hunk 2 above)
Comment (Krishnan-Priyanshu) [reply ome/resource_discovery.go]: done

File: ome/resource_discovery_schema.go [hunk 5]
```diff
@@ -283,11 +243,95 @@ func RedfishSchema() map[string]schema.Attribute {
...
+			MarkdownDescription: "Enable the Certificate Authority (CA) check.",
+			Description:         "Enable the Certificate Authority (CA) check.",
+			Optional:            true,
+			Computed:            true,
+			PlanModifiers: []planmodifier.Bool{
+				BoolDefaultValue(types.BoolValue(false)),
+			},
+		},
+
+		// "domain": schema.StringAttribute{
```
[the remark this answers is not in the document]
Comment (Krishnan-Priyanshu) [reply ome/resource_discovery_schema.go:296]: this field is under-discussion.

**Review by Krunal-Thakkar — APPROVED** [review]
LGTM

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Pull request merged**